tests/system/test_system.py

```diff
@@ -1818,8 +1850,71 @@ def test_count_query_in_transaction(client, cleanup):
 
     collection = client.collection(collection_id)
 
-    query = collection.where("a", "==", 1)
+    query = collection.where(filter=FieldFilter("a", "==", 1))
     count_query = query.count()
     result = count_query.get()
     for r in result[0]:
         assert r.value == 2  # there are still only 2 docs
+
+
+def test_query_with_and_composite_filter(query_docs):
+    collection, stored, allowed_vals = query_docs
+    and_filter = And(
+        filters=[
+            FieldFilter("stats.product", ">", 5),
+            FieldFilter("stats.product", "<", 10),
+        ]
+    )
+
+    query = collection.where(filter=and_filter)
+    for result in query.stream():
+        assert result.get("stats.product") > 5
+        assert result.get("stats.product") < 10
+
+
+def test_query_with_or_composite_filter(query_docs):
+    collection, stored, allowed_vals = query_docs
+    or_filter = Or(
+        filters=[
+            FieldFilter("stats.product", ">", 5),
+            FieldFilter("stats.product", "<", 10),
+        ]
+    )
+    query = collection.where(filter=or_filter)
+    gt_5 = 0
+    lt_10 = 0
+    for result in query.stream():
+        value = result.get("stats.product")
+        print(value)
+        assert value > 5 or value < 10
+        if value > 5:
+            gt_5 += 1
+        if value < 10:
+            lt_10 += 1
+
+    assert gt_5 > 0
+    assert lt_10 > 0
+
+
+def test_query_with_complex_composite_filter(query_docs):
```

Comment:
would be good to also add some transaction tests

Reply:
I have added system tests (sync and async) using transaction